### PR TITLE
langchain 0.3.12

### PR DESCRIPTION
**Pull request by SC426**
**Destination channel:** defaults

### Links

- [PKG-6020](https://anaconda.atlassian.net/browse/PKG-6020)
- [Upstream repository](https://github.com/langchain-ai/langchain/tree/v0.1.16)
- [Upstream changelog/diff](https://github.com/langchain-ai/langchain/releases)

### Explanation of changes:

- Update version
- Remove Snowflake abs.yaml

[PKG-6020]: https://anaconda.atlassian.net/browse/PKG-6020?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by bwingconda — COMMENTED** [review]
The dependencies to update should be [these](https://github.com/langchain-ai/langchain/blob/langchain%3D%3D0.3.12/libs/langchain/pyproject.toml#L34), right?

Could run the test suites as well.

**Comment by SC426** [comment]
@bwingconda I've updated the dependences. I feel it's better to keep the continuity between the defaults version and the snowflake version. They aren't enabling tests, so we shouldn't either.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -25,8 +25,8 @@ requirements:
   run:
     - aiohttp >=3.8.3,<4.0.0
     - async-timeout >=4.0.0,<5.0.0  # [py<311]
-    - langchain-core >=0.3.21,<0.4.0
-    - langchain-text-splitters >=0.3.0,<0.4.0
+    - langchain-core >=0.3.25,<0.4.0
+    - langchain-text-splitters >=0.3.3,<0.4.0
     - langsmith >=0.1.17,<0.2.0
```
Comment (ELundby45) [inline recipe/meta.yaml:30]: https://github.com/langchain-ai/langchain/blob/langchain%3D%3D0.3.12/libs/langchain/pyproject.toml#L38

```suggestion
    - langsmith >=0.1.17,<0.3.0
```

**Review by ELundby45 — COMMENTED** [review]
One minor change I can see, other than that it looks good to me.

File: recipe/meta.yaml (hunk 1 above)
Comment (SC426) [reply recipe/meta.yaml]: Missed that one. Thanks!

**Review by ELundby45 — APPROVED** [review]

**Review by bwingconda — APPROVED** [review]

**Pull request merged**